src/mlpack/methods/linear_regression/linear_regression_impl.hpp

```diff
@@ -209,7 +161,8 @@ LinearRegression<ModelMatType>::Train(const MatType& predictors,
   // The total runtime of this should be O(d^2 N) + O(d^3) + O(dN).
   // (assuming the SVD is used to solve it)
   arma::Mat<ElemType> cov = p * p.t() +
-      ((ElemType) lambda) * arma::eye<arma::Mat<ElemType>>(p.n_rows, p.n_rows);
+      ((ElemType) lambda.value()) *
+      arma::eye<arma::Mat<ElemType>>(p.n_rows, p.n_rows);
```

Comment:
I think this change is incorrect.  You should probably use `this->lambda` instead.